ui/apps/platform/src/Containers/Integrations/IntegrationPage.tsx

```diff
@@ -58,30 +62,37 @@ function IntegrationPage({ title, name, children }: IntegrationPageProps): React
                     <FlexItem>
                         <Title headingLevel="h1">{name}</Title>
                     </FlexItem>
-                    {pageState === 'VIEW_DETAILS' && permissions[source].write && (
-                        <FlexItem align={{ default: 'alignRight' }}>
-                            {editDisabledMessage ? (
-                                <Tooltip content={editDisabledMessage}>
+                    {pageState !== 'CREATE' &&
+                        pageState !== 'LIST' &&
+                        (type === 'generic' || type === 'splunk') && (
+                            <TraitsOriginLabel traits={traits} />
+                        )}
+                    {pageState === 'VIEW_DETAILS' &&
+                        permissions[source].write &&
+                        isUserResource(traits) && (
```

Comment:
For complicated conditions, especially the second that causes indentation and extra diff changed lines, what do you think about local variables above the `return` statement? Something like:

```ts
const hasTraitsOriginLabel = pageState !== 'CREATE' && pageState !== 'LIST' && (type === 'generic' || type === 'splunk');
const hasEditButton = pageState === 'VIEW_DETAILS' && permissions[source].write && isUserResource(traits);
```